miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java

```diff
@@ -687,10 +688,14 @@ protected LibraryDto createDtoFromParent(Sample item) {
       templatesByProjectId = results.stream()
           .map(sam -> sam.getProject().getId())
           .distinct()
-          .map(projectId -> templateService.listLibraryTemplatesForProject(projectId))
-          .filter(list -> !list.isEmpty())
-          .map(Dtos::asLibraryTemplateDtos)
-          .collect(Collectors.toMap(list -> list.get(0).getProjectId(), list -> list));
+          .map(projectId -> {
+            Map<Long, List<LibraryTemplateDto>> map = new HashMap<>();
+            map.put(projectId, Dtos.asLibraryTemplateDtos(templateService.listLibraryTemplatesForProject(projectId)));
+            return map;
+          })
+          .filter(map -> !map.values().stream().allMatch(value -> value.isEmpty()))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)